python/tk_maya/panel_generation.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+def dock_panel(engine, panel_id, widget_id, title):
+    """
+    Docks a Shotgun app panel widget in a new panel tab of Maya Channel Box dock area.
+
```

Comment:
I updated the `widget_id` param description.

No need to pass in the widget instance since only its name is needed and usable by Maya.